sentinel-adapter/sentinel-sofa-rpc-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/sofa/rpc/SentinelSofaRpcProviderFilter.java

```diff
@@ -0,0 +1,100 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.csp.sentinel.adapter.sofa.rpc;
+
+import com.alibaba.csp.sentinel.*;
+import com.alibaba.csp.sentinel.adapter.sofa.rpc.fallback.SofaRpcFallbackRegistry;
+import com.alibaba.csp.sentinel.context.ContextUtil;
+import com.alibaba.csp.sentinel.slots.block.BlockException;
+import com.alipay.sofa.rpc.common.RpcConstants;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.ext.Extension;
+import com.alipay.sofa.rpc.filter.AutoActive;
+import com.alipay.sofa.rpc.filter.FilterInvoker;
+
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getApplicationName;
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getInterfaceResourceName;
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getMethodResourceName;
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getMethodArguments;
+
+/**
+ * SOFARPC service provider filter for Sentinel, auto activated by default.
+ *
+ * If you want to disable the provider filter, you can configure:
+ * <pre>ProviderConfig.setParameter("sofa.rpc.sentinel.enabled", "false");</pre>
+ *
+ * or add setting in rpc-config.json file:
+ * <pre>
+ * {
+ *   "sofa.rpc.sentinel.enabled": false
+ * }
+ * </pre>
+ *
+ * @author cdfive
+ */
+@Extension(value = "providerSentinel", order = -1000)
+@AutoActive(providerSide = true)
+public class SentinelSofaRpcProviderFilter extends AbstractSofaRpcFilter {
+
+    @Override
+    public SofaResponse invoke(FilterInvoker invoker, SofaRequest request) throws SofaRpcException {
+        // Now only support sync invoke.
+        if (request.getInvokeType() != null && !RpcConstants.INVOKER_TYPE_SYNC.equals(request.getInvokeType())) {
+            return invoker.invoke(request);
+        }
+
+        String applicationName = getApplicationName(request);
+        String interfaceResourceName = getInterfaceResourceName(request);
+        String methodResourceName = getMethodResourceName(request);
+
+        Entry interfaceEntry = null;
+        Entry methodEntry = null;
+        try {
+            ContextUtil.enter(methodResourceName, applicationName);
+
+            interfaceEntry = SphU.entry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN);
+            methodEntry = SphU.entry(methodResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN, getMethodArguments(request));
+
+            SofaResponse response = invoker.invoke(request);
+
+            Object appResponse = response.getAppResponse();
+            if (appResponse instanceof Throwable) {
```

Comment:
According to the comments of `SofaResponse`, the `appResponse` may indicate the biz exception and the `errorMsg` indicates the framework exception. Should we handle the latter case?

Reply:
Oh, I always thought biz exception should be traced, as it often happened in actual project.
I found the commment in `Trace` class, `Tracer` is used to record other exceptions except block exception. Should we handle both of them?

Besides, I have a question why`com.alibaba.csp.sentinel.adapter.dubbo.SentinelDubboProviderFilter` doesn't catch other Exception exception `BlockException` and `RpcException`?

Reply:
> I found the comment in Trace class, Tracer is used to record other exceptions except for block exception. Should we handle both of them?

Yes, though framework error does not occur frequently, we may need to handle it.

Reply:
Got it, thanks.